### ISSUE TITLE
DNN Bias learning rate multiplier/weight decay seems to be not implemented

### ISSUE THREAD

**Issue by e-fominov** [davisking/dlib#680]
I was found that my DNN network training results are slightly different from other frameworks. When digging into code I found, that DNN convolutiuon and fc layers have separate setting for bias learning rate and weight decay, but core.h is not using them at "update_parameters" [training step](https://github.com/davisking/dlib/blob/master/dlib/dnn/core.h#L881-L893):

```C++
        template <typename solver_type>
        void update_parameters(sstack<solver_type> solvers, double learning_rate)
        {
            DLIB_CASSERT(solvers.size()>=num_computational_layers);
            // Don't try to adjust the parameters if this layer doesn't have any or the
            // learning rate is disabled for this layer.
            if (params_grad.size() != 0 && get_learning_rate_multiplier(details) != 0)
            {
                const tensor& step = solvers.top()(learning_rate, details, static_cast<const tensor&>(params_grad));
                tt::add(details.get_layer_params(), details.get_layer_params(), step);
            }
            subnetwork->update_parameters(solvers.pop(), learning_rate);
        }

```

As I see, all layer parameters are updated with the same learning rate and weight decay, no special settings for bias is used, also learning rate multiplier is used only to enable/disable training, but does not really affect learning rate of layer

Is there any implementation that I am not able to find, or its unimplemented?

**Comment by davisking** [comment]
It's in the solver object.  Look at the sgd object for instance.  This needs to be so since there is no generic way to do it if you want to be able to have different learning rates for some parts of the parameter tensor vs others.

This whole thing just reminded me that we forgot to update the solvers to apply these parameters for the new cont_ layer. :/  Need to do that.

**Comment by e-fominov** [comment]
Thank you, found it!
